### DISCUSSION TITLE
Bonsai Workflow for CSV-based Event Triggering

### DISCUSSION THREAD

**Discussion by neugun** [bonsai-rx/bonsai#1632] [Q&A]
Hi,

I am engaged in leveraging Bonsai for optogenetic stimulation, with the specific aim of reading timepoints from a CSV file and accordingly triggering optostimulation signals. The envisioned workflow entails a one-time read of the CSV data, which is then utilized to consistently trigger signals at predetermined timepoints.
![image](https://github.com/bonsai-rx/bonsai/assets/26133490/cd91359d-a76e-4e76-bff0-e40263cd2e3e)
[SoftwareStim_ZZ_V4_noncontingent.bonsai.zip](https://github.com/bonsai-rx/bonsai/files/13326899/SoftwareStim_ZZ_V4_noncontingent.bonsai.zip)

Despite my best efforts and numerous iterations, I've encountered challenges in configuring the workflow of read csv to generate output to function as required, largely due to my limited experience with compiling programs within Bonsai's framework.

Your insights into how I might rectify the issues I'm facing would be invaluable. 

Thanks a lot,
Zhenggang

**Comment by neugun** [comment]
![image](https://github.com/bonsai-rx/bonsai/assets/26133490/780800ef-bf32-460e-b3dc-7a133e6293b8)

**Reply by bruno-f-cruz** [reply]
What are you exactly struggling with? I am not sure I got the exact pattern you want help with from reading your post
